### PR TITLE
Update SaaSquatch Managed Packages for SSW Project

### PR DESCRIPTION
**Pull request by zachharrison**
## Description of the change
Adding jsDoc strings to all vanilla components and their props in preparation for the new self serve widget project.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Documentation or Development tools (readme, specs, tests, code formatting)

## Links

- Jira issue number: (PUT IT HERE)
- Process.st launch checklist: (PUT IT HERE)

### Development

- [X] [Prettier](https://www.npmjs.com/package/prettier) was run (if applicable)
- [ ] The behaviour changes in the pull request are covered by specs
- [ ] All tests related to the changed code pass in development

### Paperwork

- [x] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has a Jira number
- [ ] This pull request has a Process.st launch checklist

### Code review

- [x] Changes have been reviewed by at least one other engineer
- [x] Security impacts of this change have been considered

### CODE REVIEW

File: packages/vanilla-components/src/components/text-component/text-component.tsx [hunk 1]
```diff
@@ -2,28 +2,92 @@ import { h, Component, Prop, Element } from "@stencil/core";
...
+ */
 @Component({
   tag: "sqh-text-component",
   styleUrl: "text-component.scss",
 })
 export class TextComponent {
+  /**
+   * Hide the component
+   *
+   * @uiName Is Hidden
```
Comment (DerekSiemens) [inline packages/vanilla-components/src/components/text-component/text-component.tsx:16]: Something like `Hide Button` is more appropriate, this applies to other components as well

File: packages/vanilla-components/src/components/copy-button/copy-button.tsx [hunk 2]
```diff
@@ -3,24 +3,87 @@ import { shadeColor, addClass, removeClass } from "../../utilities";
...
+ */
 @Component({
   tag: "sqh-copy-button",
   styleUrl: "copy-button.scss",
 })
 export class CopyButton {
+   /**
+   * Hide the component
+   *
+   * @uiName Is Hidden
```
Comment (DerekSiemens) [inline packages/vanilla-components/src/components/copy-button/copy-button.tsx:17]: Something like Hide Button is more appropriate, this applies to other components as well

File: packages/vanilla-components/src/components/referral-list/referral-list.tsx [hunk 3]
```diff
@@ -3,46 +3,221 @@ import { css } from "emotion";
...
   tag: "sqh-referral-list",
   styleUrl: "referral-list.scss",
 })
 export class ReferralList {
-  // general in dropdown
+  /**
+   * Hide the component
+   *
+   * @uiName Is Hidden
+   * @uiGroup general
```
Comment (DerekSiemens) [inline packages/vanilla-components/src/components/referral-list/referral-list.tsx:18]: Props that are not meant to go in an according and are meant to be displayed at the top level will not need a uiGroup

File: packages/vanilla-components/src/components/referral-list/referral-list.tsx [hunk 4]
```diff
@@ -3,46 +3,221 @@ import { css } from "emotion";
...
   @Prop() referraltextcolor: string;
+  /**
+   * Text shown when the referred users name is unknown
+   *
+   * @uiName Unknown User Text
+   * @uiGroup general
+   */
   @Prop() unknownuser: string;
+  /**
+   * Show and hide expired referrals
```
Comment (DerekSiemens) [inline packages/vanilla-components/src/components/referral-list/referral-list.tsx:64]: Expired referrals are not a concept on our platform, I would recommend revisiting this prop and what it does. Most likely involves expired rewards from a referral

File: packages/vanilla-components/src/components/referral-list/referral-list.tsx [hunk 5]
```diff
@@ -3,46 +3,221 @@ import { css } from "emotion";
...
+   *
+   * @uiName Show Expiry
+   * @uiGroup general
+   */
   @Prop() showexpiry: boolean;
+  /**
+   * Show the invoice details for a referral
```
Comment (DerekSiemens) [inline packages/vanilla-components/src/components/referral-list/referral-list.tsx:71]: What exactly are the `details` is it the customer note on the reward given? should be a bit more detailed

File: packages/vanilla-components/src/components/referral-list/referral-list.tsx [hunk 6]
```diff
@@ -3,46 +3,221 @@ import { css } from "emotion";
...
   @Prop() noreferralsyet: string;
+  /**
+   * Text color of the referred users name
+   *
+   * @uiName Referral Name Color
+   * @uiGroup general
+   */
   @Prop() referralnamecolor: string;
+  /**
+   * Text color of the referral details
```
Comment (DerekSiemens) [inline packages/vanilla-components/src/components/referral-list/referral-list.tsx:50]: What referral details? Status, rewards, date? Needs to be more detailed

**Comment by DerekSiemens** [comment]
For the most part the JSDocs are looking good but there are some areas that still need improvement.

**Review by 00salmon — APPROVED** [review]
approved!

**Pull request merged**